frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/AddDataModal/AddDataModal.tsx

```diff
@@ -1,34 +1,75 @@
 import { useState } from "react";
 import { t } from "ttag";
 
+import { useListDatabasesQuery } from "metabase/api";
+import { useHasTokenFeature } from "metabase/common/hooks";
 import CS from "metabase/css/core/index.css";
 import { useSelector } from "metabase/lib/redux";
-import { getUserIsAdmin } from "metabase/selectors/user";
+import { getSetting } from "metabase/selectors/settings";
+import { canAccessSettings, getUserIsAdmin } from "metabase/selectors/user";
 import { Box, Icon, Modal, Tabs } from "metabase/ui";
 
 import S from "./AddDataModal.module.css";
+import { CSVPanel } from "./Panels/CSVPanel";
 import { DatabasesPanel } from "./Panels/DatabasesPanel";
 import { PanelsHeader } from "./Panels/PanelsHeader";
 import { trackAddDataEvent } from "./analytics";
+import { isValidTab } from "./utils";
 
-export const AddDataModal = ({
-  onClose,
-  opened,
-}: {
-  onClose: () => void;
+interface AddDataModalProps {
   opened: boolean;
-}) => {
+  onClose: () => void;
+}
+
+export const AddDataModal = ({ opened, onClose }: AddDataModalProps) => {
+  const { data: databaseResponse } = useListDatabasesQuery();
+
   const [activeTab, setActiveTab] = useState<string | null>("db");
 
   const isAdmin = useSelector(getUserIsAdmin);
+  const userCanAccessSettings = useSelector(canAccessSettings);
+
+  const hasAttachedDWHFeature = useHasTokenFeature("attached_dwh");
+  const databases = databaseResponse?.data;
+  const uploadDbId = useSelector(
+    (state) => getSetting(state, "uploads-settings")?.db_id,
+  );
```

Comment:
not a big deal but I would prefer `useSetting('uploads-settings');` here

Reply:
https://github.com/metabase/metabase/pull/59008/commits/2a8e803cee63af910a3dd81d231e5948374584c0